packages/stream-model-instance-handler/src/__tests__/model-instance-document-handler.test.ts

```diff
@@ -258,6 +261,29 @@ const MODEL_DEFINITION_SINGLE: ModelDefinition = {
   },
 }
 
+const MODEL_DEFINITION_SET: ModelDefinition = {
+  name: 'MyModel',
+  version: '2.0',
+  interface: false,
+  implements: [],
+  accountRelation: { type: 'set', fields: ['one', 'two'] },
```

Comment:
Hmm, I don't actually see where in this PR we update the ModelDefinition to support this.  I believe the RFC only called for supporting a single field and calling it `accountRelationField`.  I'm down to change this to take an array of fields, that seems better and more flexible.  But A) we should update the RFC and B) I'm wondering if we can make the field name more descriptive.  `accountRelationFields` would be the logical change from what's in the RFC.  I also kind of like `uniqueFields`.

FYI @rohhan

Reply:
The model definition was already updated as part of the work on interfaces, v2.0 supports the SET account relation as used here.

Reply:
in our call last week we decided to call this `accountRelationFields` I believe, instead of just `fields`. Okay if you want to do this rename in a follow-up

Reply:
I think there are 2 things here: the model structure definition (codec) and the GraphQL schema definition.
The model structure is already implemented to support `accountRelation: { type: 'set', fields: Array<string> }`, that was done along with the interfaces change to update the model definition version from v1.0 to v2.0.
Regarding the GraphQL schema, we have the `@createIndex(description: String!, accountRelation: ModelAccountRelation)` that needs to be updated to `@createIndex(description: String!, accountRelation: ModelAccountRelation, accountRelationFields: [String!])`, but that's on the ComposeDB tooling side, that itself creates the models from the GraphQL schema.
So basically at a high level yes we'll have `accountRelationFields`, but internally for the model structure these fields are specified along with the relation `type` in the `accountRelation` object.

Reply:
The model streamtypes are still part of the public API.  Especially with things like OrbisDB coming online, not everyone is going to be going through the ComposeDB tooling.

Should be fine to change the model definition right?  Not like anyone can be dependent on it yet.

Reply:
Yes we can change it but it's a pretty heavy change, it requires updating lots of types and handlers in ComposeDB to support the format change as a new major version of the model definition.

Having the `fields` along the `type` in the `accountRelation` object was intended from the beginning, that's why we changed `accountRelation` to be an object rather than a string when we started using codeco, so we can strongly type any additional information related to the account relation type.

Basically, with the current format we have something like:
```ts
type AccountRelationV2 = { type: 'list' } | { type: 'single' } | { type: 'set', fields: Array<string> }

type ModelDefinitionV2 = {
  version: '2.0',
  description: string,
  accountRelation: AccountRelationV2,
  // ... other fields
}
```
If we moved the `accountRelationFields` to the root, we'd need something like:
```ts
type ModelRelationV2 =
| {
  version: '2.0',
  description: string,
  accountRelation: { type: 'list' | 'single' },
  // ... other fields
}
| {
  version: '2.0',
  description: string,
  accountRelation: { type: 'set' },
  accountRelationFields: Array<string>,
  // ... other fields
}
```
It doesn't make sense to me to extract the `accountRelationFields` from the `accountRelation` object, it adds complexity without bringing any benefit IMO.

Reply:
Yep, that makes sense.  Sorry, I didn't notice that this was bundled within the accountRelation object.  Given that, from context it's clear that "fields" applies to the account relation, no need to rename it.  And definitely agree we don't want to move it up to the top level.  I think I wasn't fully processing the distinction between the `@createModel` directive semantics from the RFC and the Model definition structure.  Thanks for clarifying.